portals/devportal/src/main/webapp/services/login/login_callback.jsp

```diff
@@ -84,8 +84,20 @@
         postLogoutRedirectURI = serverUrl + appContext + LOGOUT_CALLBACK_URL_SUFFIX;
     }
     String error = request.getParameter("error");
-    if (error != null && error.equals("login_required")) {
-        response.sendRedirect(postLogoutRedirectURI + "?referrer=" + referrer);
+    boolean isAnonymousEnabled = false;
+    if ((error != null) && error.equals("login_required")) {
+        if(state!=null){
```

Comment:
Formatting error, please check other places as well.

Reply:
Fixed in https://github.com/wso2/apim-apps/pull/523